examples/nlp/semantic_similarity_with_bert.py

```diff
@@ -0,0 +1,363 @@
+"""
+Title: Semantic Similarity with BERT
+Author: [Mohamad Merchant](https://twitter.com/mohmadmerchant1)
+Date created: 2020/08/15
+Last modified: 2020/08/20
+Description: Natural Language Inference by Fine-tuning BERT model on SNLI Corpus.
+"""
+"""
+## Introduction
+
+Semantic Similarity or Natural Languauge Inference is the task of determining how two
+sentences are similar to each other in terms of their meaning.
+This example demonstrates the use of SNLI (Standford Natural Language Inference) Corpus
+to perform semantic similarity with Transformers.
+We will fine-tune BERT model by providing two sentences as inputs and it outputs the
+probability of the similarity between sentences.
+
+### References
+* [BERT](https://arxiv.org/pdf/1810.04805.pdf)
+* [SNLI](https://nlp.stanford.edu/projects/snli/)
+"""
+
+"""
+## Setup
+"""
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+import transformers
+from transformers import BertTokenizer, TFBertModel
+
+"""
+## Configurations
+"""
+
+max_length = 128  # Maximum length of input sentence to the model.
+batch_size = 32
+epochs = 4
+learning_rate = 3e-5
+np.random.seed(42)
```

Comment:
It is unclear what this seed is meant to control. It will have no effect over Keras reproducibility. If it is specifically for the data shuffling, prefer passing a `seed` argument to your `Sequence` object and use it locally there (via a `RandomState` object). In general avoid global seeding since it has side effects.